src/constitutive/TACSConstitutive.h

```diff
@@ -134,6 +139,30 @@ class TACSConstitutive : public TACSObject {
                                 const double pt[], const TacsScalar X[],
                                 int dvLen, TacsScalar dfdx[]) {}
 
+  virtual void addDensityDVSens(int elemIndex, TacsScalar scale,
+                                const double pt[], const TacsScalar X[],
+                                int dvLen, TacsScalar dfdx[],
+                                const TacsScalar u[]) {
```

Comment:
We might want to change the state variable name from `u` to `vars` to be consistent with the rest of TACS?